In [1]:
import nsfg
import math

In [2]:
data = nsfg.ReadFemPreg()#load dataset
live = data[data.outcome == 1] #live births only

In [3]:
(live.shape[0]/data.shape[0])*100 #represents around 67% of all data.

67.29934525123225

In [4]:
first = live[live.birthord == 1]
others = live[live.birthord != 1]

In [5]:
type(first)

pandas.core.frame.DataFrame

In [6]:
type(others)

pandas.core.frame.DataFrame

In [11]:
def CohenEffectSize(group1, group2):
    '''
    If I was passing on the result of this function, I would return d.
    
    For interpretability, I included print statements.
    
    Inputs:
        - group1, group2: Pandas Dataframes
    '''
    diff = group1.mean() - group2.mean()
    
    print("Group 1 Mean is {}".format(group1.mean()))
    print()
    print("Group 2 Mean is {}".format(group2.mean()))
    print()
    print("The difference is {}".format(diff))
    print()
    
    var1 = group1.var()
    var2 = group2.var()
    
    n1 = len(group1)
    n2 = len(group2)
    
    pooled_var = (n1 * var1 + n2 * var2)/(n1 + n2)
    
    d = diff / math.sqrt(pooled_var)
    
    print("The effect-size is: %f" % d)

In [12]:
CohenEffectSize(first.totalwgt_lb, others.totalwgt_lb)

Group 1 Mean is 7.201094430437772

Group 2 Mean is 7.325855614973262

The difference is -0.12476118453549034

The effect-size is: -0.088673


From the above, we can see that first babies are lighter than those babies that are not first.

However, the difference here is trivial as the two groups means diffe by a standard deviation of $-0.088673$.

Cohen suggested that:
1. $d=0.2$ could be considered a small effect

2. $d=0.5$ could be considered a medium effect

3. $d=0.8$ could be considered a large effect.

Since our difference is $<0.2$, it can be considered trivial, even if we found it to be statistically significant